### DISCUSSION TITLE
How to export named interface in generated bindings.ts

### DISCUSSION THREAD

**Discussion by ofofofofo** [oscartbeaumont/rspc#119] [Specta Help] [answered]
When I have a Struct 
```
#[derive(Serialize, Type)]
pub struct User {
    pub name: String,
}
```
I want specta generate interface like this  

`export interface User { name: string; }`

instead of anonymous type 
```
    | {
        key: "me";
        input: null;
        result: {
          name: string;
        };
      }
```

**Comment by oscartbeaumont** [comment, marked as answer]
You shouldn't be relying on the format of the rspc bindings because they may change between releases. I need to mention this more in the documentation. You can do something like this in your code.

```ts
import { inferProcedureResult } from "@rspc/client";
import { Procedures } from "./my-bindings-from-rust.ts";

export type User = inferProcedureResult<Procedures, "query", "me">;
```

**Reply by oscartbeaumont** [reply]
You have put this in the Specta category which is confusing me because the anonymous type you showed is in the rspc format. If your goal is to export the type directly with Specta you can refer to there docs [here](https://docs.rs/specta/latest/specta/).
